src/components/movie/movie-form.tsx

```diff
@@ -1,111 +1,160 @@
-            ? e.target.value
-            : e.target.valueAsNumber || 0;
+    return (
+        <Form
+            onSubmit={onSubmit}
+            initialValues={movie}
+            subscription={{ submitting: true, pristine: true }}
+            render={({ handleSubmit, form, submitting }) => (
+                <form onSubmit={handleSubmit}>
+                    <div className="form-grid">
+                        <Field name="title" validate={required}>
+                            {({ input, meta }) => (
+                                <div className="form-group">
+                                    <label>Title</label>
+                                    <input
+                                        {...input}
+                                        type="text"
+                                        className="form-control"
+                                        placeholder="Title"
+                                    />
+                                    {meta.error && meta.touched && <span>{meta.error}</span>}
+                                </div>
+                            )}
+                        </Field>
 
-        onChange(key, value);
-    }
+                        <Field name="release_date">
+                            {({ input }) => (
+                                <div className="form-group">
+                                    <label>RELEASE DATE</label>
+                                    <input
+                                        {...input}
+                                        type="text"
+                                        className="form-control"
+                                        placeholder="Select date"
+                                    />
+                                </div>
+                            )}
+                        </Field>
 
-    const onSelect = (options: string[]) => {
-        onChange('genres', options);
-    }
+                        <Field name="poster_path" validate={required}>
+                            {({ input, meta }) => (
+                                <div className="form-group">
+                                    <label>movie url</label>
+                                    <input
+                                        {...input}
+                                        type="text"
+                                        className="form-control"
+                                        placeholder="http://"
+                                    />
+                                    {meta.error && meta.touched && <span>{meta.error}</span>}
+                                </div>
+                            )}
+                        </Field>
 
-    const { title, genres, poster_path, runtime, overview, release_date, vote_average } = movie;
+                        <Field
+                            name="vote_average"
+                            validate={composeValidators(mustBeNumber, minValue(0), maxValue(10))}
```

Comment:
After submitting you send value as string, but backend validation said that it should be a number, so you need to pass it as number. This is `final-form` default behavior to send all data in string format, but you can change it by add prop [`parse`](https://final-form.org/docs/react-final-form/types/FieldProps#:~:text=about%20Field%20Names.-,parse,to%20parse%2C%20thus%20allowing%20you%20to%20have%20form%20values%20of%20%27%27.,-ref) to fix it